nixos/modules/services/x11/xserver.nix

```
@@ -591,8 +591,23 @@ in
       let dmconf = cfg.displayManager;
           default = !(dmconf.gdm.enable
                     || dmconf.sddm.enable
-                    || dmconf.xpra.enable );
-      in mkIf (default) true;
+                    || dmconf.xpra.enable
+                    || dmconf.sx.enable
+                    || dmconf.startx.enable);
+      in mkIf (default) (mkDefault true);
+
+    # so that the service won't be enabled when only startx is used
+    systemd.services.display-manager.enable  =
+      let dmconf = cfg.displayManager;
+          nodmUsed = !(dmconf.gdm.enable
+                    || dmconf.sddm.enable
+                    || dmconf.xpra.enable
+                    || dmconf.lightdm.enable);
+      in mkIf (nodmUsed) (mkDefault false);
```

```suggestion
                    || dmConf.xpra.enable
                    || dmConf.sx.enable
                    || dmConf.startx.enable);
      in mkIf (default) (mkDefault true);

    # so that the service won't be enabled when only startx is used
    systemd.services.display-manager.enable  =
      let dmConf = cfg.displayManager;
          noDmUsed = !(dmconf.gdm.enable
                    || dmConf.sddm.enable
                    || dmConf.xpra.enable
                    || dmConf.lightdm.enable);
      in mkIf (noDmUsed) (mkDefault false);
```
lets use camelCase to make it easier to read.